### ISSUE TITLE
ConcurrentModificationException while removing ChangeListener

### ISSUE THREAD

**Issue by StefMa** [bluelinelabs/Conductor#385]
I have the following code:
```java
        getRouter().addChangeListener(new ControllerChangeListener() {
            @Override
            public void onChangeCompleted(@Nullable final Controller to, @Nullable final Controller from,
                    final boolean isPush,
                    @NonNull final ViewGroup container, @NonNull final ControllerChangeHandler handler) {
                // We are going to be removed... Remove Presenter
                if (!isPush && from == CustomController.this) {
                  // Do stuff
                }
                getRouter().removeChangeListener(this);
            }

            @Override
            public void onChangeStarted(@Nullable final Controller to, @Nullable final Controller from,
                    final boolean isPush,
                    @NonNull final ViewGroup container, @NonNull final ControllerChangeHandler handler) {

            }
        });
```

Unfortunately it crashes with `ConcurrentModificationException` if I rotate the device and press back:
```
11-22 10:08:35.966 5958-5958/a.package.name.dev E/AndroidRuntime: FATAL EXCEPTION: main
                                                                             Process: a.package.name.dev, PID: 5958
                                                                             java.util.ConcurrentModificationException
                                                                                 at java.util.ArrayList$ArrayListIterator.next(ArrayList.java:573)
                                                                                 at com.bluelinelabs.conductor.ControllerChangeHandler$1.onChangeCompleted(ControllerChangeHandler.java:216)
                                                                                 at com.bluelinelabs.conductor.changehandler.SimpleSwapChangeHandler.performChange(SimpleSwapChangeHandler.java:78)
                                                                                 at com.bluelinelabs.conductor.ControllerChangeHandler.executeChange(ControllerChangeHandler.java:204)
                                                                                 at com.bluelinelabs.conductor.ControllerChangeHandler.executeChange(ControllerChangeHandler.java:154)
                                                                                 at com.bluelinelabs.conductor.Router.performControllerChange(Router.java:774)
                                                                                 at com.bluelinelabs.conductor.Router.performControllerChange(Router.java:744)
                                                                                 at com.bluelinelabs.conductor.Router.performControllerChange(Router.java:726)
                                                                                 at com.bluelinelabs.conductor.Router.popController(Router.java:136)
                                                                                 at a.package.name.presentation.players.detail.PlayerDetailController$onCreateView$1.onClick(PlayerDetailController.kt:104)
                                                                                 at android.view.View.performClick(View.java:4780)
                                                                                 at android.view.View$PerformClick.run(View.java:19866)
                                                                                 at android.os.Handler.handleCallback(Handler.java:739)
                                                                                 at android.os.Handler.dispatchMessage(Handler.java:95)
                                                                                 at android.os.Looper.loop(Looper.java:135)
                                                                                 at android.app.ActivityThread.main(ActivityThread.java:5254)
                                                                                 at java.lang.reflect.Method.invoke(Native Method)
                                                                                 at java.lang.reflect.Method.invoke(Method.java:372)
                                                                                 at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
                                                                                 at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

Do I need to remove the change listener?
Or isn't it needed?

**Comment by gstepniewski** [comment]
Where are you doing this? I tried putting your snipper in both Controller and Activity but could not reproduce the crash.

Also, not sure if you actually need to keep adding and removing the same Listener over and over... Especially if it doesn't even have parameters and always does the same thing!

**Comment by EricKuck** [comment]
Sorry for the very late response, but this should have been fixed a while ago, with 66b0458c. If you still have issues feel free to reopen.
